File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/options/FetchChannelsOptions.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;

public class FetchChannelsOptions {

    @Nullable
    private final Integer limit;

    @Nullable
    private final Integer offset;

    @Nullable
    private final String query;

    public FetchChannelsOptions(@NonNull PluginCall call) {
        this.limit = call.getInt("limit");
        this.offset = call.getInt("offset");
        this.query = call.getString("query");
    }

    @Nullable
    public Integer getLimit() {
        return limit;
    }

    @Nullable
    public Integer getOffset() {
        return offset;
    }

    @Nullable
    public String getQuery() {
        return query;
    }
}

File: packages/live-update/ios/Plugin/Classes/Options/FetchChannelsOptions.swift
import Foundation
import Capacitor

@objc public class FetchChannelsOptions: NSObject {
    private var limit: Int?
    private var offset: Int?
    private var query: String?

    init(_ call: CAPPluginCall) {
        self.limit = call.getInt("limit") as Int?
        self.offset = call.getInt("offset") as Int?
        self.query = call.getString("query")
    }

    func getLimit() -> Int? {
        return limit
    }

    func getOffset() -> Int? {
        return offset
    }

    func getQuery() -> String? {
        return query
    }
}
